packages/browser-lite/src/schema_validator.ts

```diff
@@ -0,0 +1,67 @@
+import * as ajv from 'ajv';
+
+export interface SchemaValidationResult {
+    success: boolean;
+    errors: string[];
+    fatal?: string;
+}
+
+export interface SchemaValidator {
+    orderValidator: (input: string) => SchemaValidationResult;
+    messageValidator: (input: string) => SchemaValidationResult;
+}
+
+/**
+ * Sets up a schema validator on the window object.
+ * @param customOrderSchema The custom filter that will be used by the root
+ *        schemas.
+ * @param schemas These are all of the schemas that can be compiled before the
+ *        customOrderSchema.
+ * @param rootSchemas The root schemas. These must be compiled last.
+ */
+export function setSchemaValidator(customOrderSchemaString: string, schemas: string[], rootSchemas: string[]): void {
+    // NOTE(jalextowle): This causes `ajv` to look at `id` fields rather than
+    // `$id` fields.
+    const AJV = new ajv();
+    for (const schema of schemas) {
+        AJV.addSchema(JSON.parse(schema));
+    }
+    const customOrderSchema = JSON.parse(customOrderSchemaString);
+    AJV.addSchema({
+        ...customOrderSchema,
+        $id: '/customOrder',
+    });
+    for (const schema of rootSchemas) {
+        AJV.addSchema(JSON.parse(schema));
+    }
+
+    const orderValidate = AJV.getSchema('/rootOrder');
+    if (orderValidate === undefined) {
+        throw new Error('Cannot find "/rootOrder" schema in AJV');
+    }
+    const messageValidate = AJV.getSchema('/rootOrderMessage');
+    if (messageValidate === undefined) {
+        throw new Error('Cannot find "rootOrderMessage" schema in AJV');
+    }
+    (window as any).schemaValidator = {
+        orderValidator: constructValidationFunctionWrapper(orderValidate),
+        messageValidator: constructValidationFunctionWrapper(messageValidate),
+    };
+}
+
+function constructValidationFunctionWrapper(
+    validationFunction: ajv.ValidateFunction,
+): (input: string) => SchemaValidationResult {
+    return (input: string) => {
+        const result: any = { success: false, errors: [] };
```

Comment:
Can you do a find all search for `: any`? There should be no reason to use it except when assigning to window via `(window as any)`.

I tried making this change:

```suggestion
        const result = { success: false, errors: [] };
```

but it causes type errors due to the return type of `validationFunction`. I think this is a potential bug in the code. It's either that or the type definitions for `ajv.ValidateFunction` are wrong. In either case, setting the type to `any` hides the problem and should be avoided in general.

<img width="703" alt="Screen Shot 2020-05-26 at 8 36 44 PM" src="https://user-images.githubusercontent.com/800857/82975217-b0e31300-9f90-11ea-8f69-3627ae49f5cc.png">

Reply:
I was able to address several areas where we use `as any` in `packages/`. There are still several lingering `as any` casts in the `@0x/mesh-rpc-client` package. I'll address these in a follow-up PR if you're alright with this.